modules/nf-scil/reconst/fodf/main.nf

```diff
@@ -55,20 +74,34 @@ process RECONST_FODF {
     export OMP_NUM_THREADS=1
     export OPENBLAS_NUM_THREADS=1
 
-    scil_extract_dwi_shell.py $dwi $bval $bvec $fodf_shells \
-        dwi_fodf_shells.nii.gz bval_fodf_shells bvec_fodf_shells \
-        $dwi_shell_tolerance -f
+    if [ "$set_method" = "ssst" ]
+    then
+
+        scil_dwi_extract_shell.py $dwi $bval $bvec $fodf_shells \
+            dwi_fodf_shells.nii.gz bval_fodf_shells bvec_fodf_shells \
+            $dwi_shell_tolerance -f
+
+        scil_fodf_ssst.py dwi_fodf_shells.nii.gz bval_fodf_shells bvec_fodf_shells $wm_frf ${prefix}__fodf.nii.gz \
+            $sh_order $sh_basis --b0_threshold $b0_thr_extract_b0 \
+            $set_mask $processes
+
+    elif [ "$set_method" = "msmt" ]
+    then
+
+        scil_fodf_msmt.py $dwi $bval $bvec $wm_frf $gm_frf $csf_frf \
+            $sh_order $sh_basis $set_mask $processes $dwi_shell_tolerance \
+            --not_all $wm_fodf $gm_fodf $csf_fodf $vf $vf_rgb
 
-    scil_compute_ssst_fodf.py dwi_fodf_shells.nii.gz bval_fodf_shells bvec_fodf_shells $frf ${prefix}__fodf.nii.gz \
-        $sh_order $sh_basis --force_b0_threshold \
-        $set_mask $processes
+        cp ${prefix}__wm_fodf.nii.gz ${prefix}__fodf.nii.gz
+
+    fi
 
     if $run_fodf_metrics
     then
 
-        scil_compute_fodf_max_in_ventricles.py ${prefix}__fodf.nii.gz $fa $md \
```

Comment:
For msmt, it simply gives a warning and outputs 0, so the final value for a_threshold ends up being the "default" 1E-10 (in the bash code below). I tought this was fine, because if you ever have "ugly" msmt-fodfs, you might get something in the ventricles. If not, then 1E-10 is used and the r_threshold becomes the most important threshold.